src/state/electrumX/sagas.ts

```diff
@@ -175,10 +175,21 @@ export function* checkConnection() {
     const currentIsServerConnectedSelector = yield select(isServerConnectedSelector);
 
     const { internetState, isServerConnected } = yield all({
-      internetState: call(() => NetInfo.fetch()),
+      internetState: call(() =>
+        NetInfo.fetch().then((state: NetInfoState) => {
+          if (state.isInternetReachable === null) return;
+          if (state.isInternetReachable !== undefined) {
+            return true;
+          }
+        }),
+      ),
       isServerConnected: call(() => BlueElectrum.ping()),
     });
-    const { isInternetReachable } = internetState;
+    if (internetState === undefined) {
+      throw new Error('Error');
+    }
+
+    const isInternetReachable = internetState;
```

Comment:
should be `const {isInternetReachabl} = internetState;` ? if not, what's the point?

Reply:
done